account_payment_purchase/models/account_move.py

```diff
@@ -37,4 +36,5 @@ def _onchange_purchase_auto_complete(self):
             return res
         elif self.partner_bank_id.id != new_bank:
             self.partner_bank_id = new_bank
+        res = super()._onchange_purchase_auto_complete()
```

Comment:
Why this change not commented anywhere?

Reply:
Do you mean commenting it in the code? The reason of changing the order of super is because when it is called, the payment mode of invoice changes and then self.payment_mode_id and new_mode are the same at the point it is checked. If you check "_onchange_purchase_auto_complete" method in purchase, it calls the _prepare_invoice and changes the payment mode, so it was not returning a warning when new mode was different from the previous one.

Reply:
I mean that you are not mentioning this change, its purpose, and so on, and it's totally unrelated to the main purpose of this PR. Remove it, and you can propose it in another one.

Reply:
But I mean that it is related to the change I have made in _prepare_invoice method to propagate the payment mode, in my opinion it is not an independent PR. Previously, it was correct because when the payment mode was different a warning was returned, but now with the change in _prepare_invoice, it is required to change the order of the super or the payment mode of invoice will change before checking if it is the different or not to the previous one.

Reply:
But the idea is to change it and show the warning. With this change, there will never be any warning...

Reply:
Yes, as the lines are added as they were before. You just receive the warning.